node-jose/index.d.ts

```diff
@@ -0,0 +1,1380 @@
+     * @property {Function} recipient A function called after a valid key is
+     *           determined; it takes an object describing the recipient, and
+     *           returns a Promise that is fulfilled once the handler's processing
+     *           is complete.
+     * @property {Function} prepare A function called just prior to decrypting
+     *           the ciphertext; it takes an object describing the decryption
+     *           result (but containing `ciphertext` and `tag' instead of `payload`
+     *           and `plaintext`), and returns a Promise that is fulfilled once
+     *           the handler's processing is complete.
+     * @property {Function} complete A function called once decryption is complete,
+     *           just prior to fulfilling the Promise returned by `decrypt()`; it
+     *           takes the object that will be returned by `decrypt()`'s fulfilled
+     *           Promise, and returns a Promise that is fulfilled once the handler's
+     *           processing is complete.
+     */
+    type decryptCritProcessors = {
+        recipient?: Function,
+        prepare?: Function,
+        complete?: Function
+    };
+
+    /**
+     * Object that defines the output that is given from the `decrypt()` function
+     * of a `Decryptor`.
+     *
+     * @property {any} header The JOSE Header, combined from the relevant "header" and
+     *           "protected" fields from the original JWE object.
+     * @property {string[]} protected An array containing the names of the protected fields
+     * @property {JWK.Key} key The used to decrypt the content
+     * @property {Buffer} payload The decrypted content (as a Buffer)
+     * @property {Buffer} plaintext An alias for `payload`
+     */
+    type decryptReturn = {
+        header: any,
+        protected: string[],
+        key: JWK.Key,
+        payload: Buffer,
+        plaintext: Buffer
+    };
+
+    /**
+     * The options for {@link createEncrypt}.
+     *
+     * @property {boolean|string} zip Determines the compression algorithm to
+     *           apply to the plaintext (if any) before it is encrypted. This can
+     *           also be `true` (which is equivalent to `"DEF"`) or **`false`**
+     *           (the default, which is equivalent to no compression).
+     * @property {string} format Determines the serialization format of the
+     *           output.  Expected to be `"general"` for general JSON
+     *           Serialization, `"flattened"` for flattened JSON Serialization,
+     *           or `"compact"` for Compact Serialization (default is
+     *           **`"general"`**).
+     * @property {boolean} compact Determines if the output is the Compact
+     *           serialization (`true`) or the JSON serialization (**`false`**,
+     *           the default).
+     * @property {string} contentAlg The algorithm used to encrypt the plaintext
+     *           (default is **`"A128CBC-HS256"`**).
```

Comment:
it's not clear whether `contentAlg` is open-ended, but even if it is `contentAlg: "A128CBC-HS256" | string` is better documentation than just string, even though they are technically equivalent.